alibi_detect/cd/base.py

```diff
@@ -746,7 +746,8 @@ def __init__(
             self,
             x_ref: Union[np.ndarray, list],
             p_val: float = .05,
-            preprocess_x_ref: bool = True,
+            x_ref_preprocessed: Optional[bool] = False,
+            preprocess_at_init: Optional[bool] = True,
```

Comment:
Technically `Optional[X]` is equivalent to `Union[X, None]` (https://docs.python.org/3/library/typing.html#typing.Optional), but here `None` is not valid and there is a default non-null value, so I would remove the `Optional` hints.

Reply:
oof I realized this (and corrected) for the new detectors PR but missed these. Will remove `Optional` now. Thanks!